addons/binding/org.openhab.binding.edimax/src/main/java/org/openhab/binding/edimax/internal/commands/AbstractCommand.java

```diff
@@ -0,0 +1,360 @@
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.xml.bind.DatatypeConverter;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.openhab.binding.edimax.internal.ConnectionInformation;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+/**
+ * Base class for commands.
+ *
+ * @author Heinz
+ *
+ * @param <T>
+ */
+public abstract class AbstractCommand<T extends Object> {
+
+    /**
+     * GET constructor.
+     */
+    public AbstractCommand() {
+        // default
+    }
+
+    /**
+     * SET constructor.
+     *
+     * @param newValue
+     */
+    public AbstractCommand(T newValue) {
+        setValue = newValue;
+    }
+
+    protected String getCommandString() {
+        List<String> list = getPath();
+        StringBuffer command = new StringBuffer();
+        command.append(XML_HEADER);
+        recurseSubList(list, command);
+        return command.toString();
+    }
+
+    protected void recurseSubList(List<String> list, StringBuffer command) {
+        String element = list.get(0);
+
+        if (list.size() == 1) {
+            // basket case
+            command.append(createLeafTag(element));
+            return;
+        }
+
+        // usually.
+        command.append(createStartTag(element));
+        List<String> subList = list.subList(1, list.size());
+        recurseSubList(subList, command);
+        command.append(createEndTag(element));
+    }
+
+    /**
+     * Overwrite and add your path entry.
+     *
+     * @return
+     */
+    protected List<String> getPath() {
+        ArrayList<String> list = new ArrayList<>();
+        list.add("SMARTPLUG");
+        return list;
+    }
+
+    /**
+     * Returns XPath expression to load response for a get request.
+     *
+     * @return
+     */
+    protected String getXPathString() {
+        List<String> list = getPath();
+        StringBuffer sb = new StringBuffer();
+        for (int i = 0; i < list.size(); i++) {
+            sb.append("/" + list.get(i));
+        }
+        sb.append("/text()");
+        return sb.toString();
+    }
+
+    /**
+     * Format usually used for dates.
+     */
+    private SimpleDateFormat format = new SimpleDateFormat("yyyyMMddkkmmss");
```

Comment:
1) Please note that SimpleDateFormat is not thread-safe, in java 8 [DateTimeFormatter](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html) which is both thread-safe and immutable.
2) Please group all the fields at the top of your class.